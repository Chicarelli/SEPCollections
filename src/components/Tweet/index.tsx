import { ReactElement } from "react";
import {
  Container,
  Avatar,
  Content
} from './styles';


interface TweetProps {
  img?: string;
  name: string;
  username: string;
  text: string;
}

const Tweet = ({img, name, username, text}: TweetProps): ReactElement => {
  return (
    <Container>
      <Avatar></Avatar>
      <Content>

      </Content>
    </Container>
  )
}

export default Tweet;