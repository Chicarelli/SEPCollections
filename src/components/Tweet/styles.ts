import styled from 'styled-components';

export const Container = styled.section`
  width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  background: ${({theme}) => theme.colors.green[800]};
`;

export const Avatar = styled.div`
  width: 60px;
  height: 60px;
  background: ${props => props.theme.colors.bege.main};
  border-radius: 100%;
`;

export const Content = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  color: #FFF;
  margin-left: .7rem;
  padding-top: .5rem;

  span {
    color: #B1B1B1;
    font-weight: bold;
  }

  article {
    margin-top: 10px;
  }
`;