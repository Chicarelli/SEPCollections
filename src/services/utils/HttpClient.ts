import axios, { AxiosInstance } from 'axios';

interface IHttpClient {
  baseURL: string|undefined;
  config?: {};
}

class HttpClient {
  protected readonly instance: AxiosInstance;

  constructor({baseURL, config}: IHttpClient) {
    this.instance = axios.create({
      baseURL,
      ...config
    });
  }

  async get(path: string) {
    const response = await this.instance.get(path);

    return response.data;
  }

  async post(path: string, data: {}) {
    const response = await this.instance.post(path, data);

    return response;
  }

}

export default HttpClient;