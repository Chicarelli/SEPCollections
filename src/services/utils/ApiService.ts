import HttpClient from './HttpClient';

export interface IListTweets {
  author_id: string;
  id: string;
  text: string;
  user: {
    id: string;
    name: string;
    username: string;
    description: string;
    profile_image_url: string;
  }
} 

class ApiService {
  httpClient: HttpClient;

  constructor() {
    this.httpClient = new HttpClient({
      baseURL: process.env.NEXT_PUBLIC_API_URL
    })
  }

  async listTweets() {
    const response = await this.httpClient.get(`tweets`);

    return response;
  }
}

export default new ApiService;