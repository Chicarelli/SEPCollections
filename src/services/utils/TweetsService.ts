import { AxiosResponse } from 'axios';
import HttpClient from "./HttpClient";

export interface IListTweets {
  author_id: string;
  id: string;
  text: string;
  user: {
    id: string;
    name: string;
    username: string;
    description: string;
    profile_image_url: string;
  }
} 
class TweetsService {
  httpClient: HttpClient;

  constructor() {
    this.httpClient = new HttpClient({baseURL: 'http://localhost:8000'});
  }

  async listTweets(): Promise<IListTweets[]> {
    const response: IListTweets[] = await this.httpClient.get('/latest-posts');

    return response;
  }
}

export default new TweetsService();