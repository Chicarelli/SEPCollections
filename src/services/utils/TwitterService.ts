import { AxiosResponse } from 'axios';
import HttpClient from "./HttpClient";

interface ITwitterPost {
  author_id: string;
  text: string;
  id: string;
}

interface ITwitterUser {
  id: string;
  name: string;
  username: string;
  description: string;
  profile_image_url: string;
}

interface TwitterResponse {
  data: ITwitterPost[],
  includes: {
    users: ITwitterUser[]
  },
  meta: {
    newest_id: string;
    oldest_id: string;
    result_count: number;
    next_token: string;
  }
}

class TwitterService {
  httpClient: HttpClient;

  constructor() {
    this.httpClient = new HttpClient({baseURL: 'https://api.twitter.com/2', config: {
      headers: {
        Authorization: `Bearer ${process.env.TWITTER_TOKEN}`,
      }
    }});
  }

  async listTweets(){
    const params = {
      max_results: '100',
      query: 'palmeiras',
      'user.fields': 'description,profile_image_url',
      expansions: 'author_id'
    }

    const searchParams = new URLSearchParams(params);
    
    const response: TwitterResponse = await this.httpClient.get(`tweets/search/recent?${searchParams.toString()}`);

    return response;
  }
}

export default new TwitterService();