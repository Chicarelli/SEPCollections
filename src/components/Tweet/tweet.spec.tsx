import { render } from '../../test/utils';
import Tweet from './index';

describe('component Tweet', () => {

  it('renders correctly', () => {
    const { getByText } = render(
      <Tweet 
        id="1" 
        name="John Doe" 
        username="johndoe" 
        text="Twitter Content" 
        img={process.env.IMAGE_URL || ""}
        />
      );
    
      expect(getByText('Twitter Content')).toBeTruthy();
  })

  it('renders correctly if there is no profile image', () => {
    const { queryByAltText } = render(
      <Tweet 
        id="1" 
        name="John Doe" 
        username="johndoe" 
        text="Twitter Content" 
        img=""
      />
    );

    expect(queryByAltText("profile")).toBeNull();
  })


  it('is link correct', () => {
    const element = render(
      <Tweet
        id="1" 
        name="John Doe" 
        username="johndoe" 
        text="Twitter Content" 
        img=""
      />
    );
    const link = element.getByRole('link')
  
    expect(link).toMatchSnapshot();
  })

})
