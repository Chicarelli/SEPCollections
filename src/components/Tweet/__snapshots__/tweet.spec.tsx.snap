// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link is correct 1`] = `
<a
  aria-label="Confira este tweet no twitter"
  href="https://twitter.com/johndoe/status/1"
  target="_blank"
>
  John Doe
   
  <span>
    @
    johndoe
  </span>
</a>
`;
